docs/cookbook/images/images-on-entity.rst

```diff
@@ -313,23 +313,20 @@ Register the form extension as a service:
             tags:
                 - { name: form.type_extension, extended_type: Sylius\Bundle\ShippingBundle\Form\Type\ShippingMethodType }
 
-11. Override the definition of the ImageUploader service
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+11. Override the definition of the ImagesUploadListener service
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 In order to handle the image upload you need to attach the image upload listener to the ``ShippingMethod`` entity events:
 
 .. code-block:: yaml
 
     # services.yml
     services:
-        sylius.listener.image_upload:
+        app.listener.image_upload:
             class: Sylius\Bundle\CoreBundle\EventListener\ImagesUploadListener
+            decorates: sylius.listener.image_upload
```

Comment:
`tags` are not inherited from the parent service. Or do you mean we should use `decorates` together with `parent`?

Reply:
I don't think it'll work: https://github.com/symfony/symfony/issues/14543

Reply:
IIRC `tags` aren't inherited from the decorated service as well (for this PR, these product & taxon tags are still assigned to `@app.listener.image_upload.inner` and shipping method ones will be assigned to `@app.listener.image_upload`).

Reply:
Tags are copied over when decorating. See the referenced PR in the description. :)

Reply:
Anyway, since we use decoration, anywhere `sylius.listener.images_upload` is used, it'll basically be replaced by `app.listener.images_upload`. The original service will not be used anymore. In fact, it will be automatically removed from the container if it's a private service and unused.

Reply:
Oh, and I just noticed the original typo in the service name (or it was changed). The free extra benefit we get by using decoration is that it'll fail if the service being decorated is not found. :smiley: